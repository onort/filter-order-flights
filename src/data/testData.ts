import { StopsType } from "../types"

export default [
  {
    carriers: [
      {
        code: "PC",
        displayCode: "PC",
        id: 1161,
        imgUrl: "http://s1.apideeplink.com/images/airlines/H9.png",
        name: "Pegasus Airlines"
      }
    ],

    destination: {
      airportCode: "ADB",
      time: "2017-04-14T09:10:00"
    },
    duration: 90,
    id: "9249-1704140740--31973-0-9250-1704140910",
    legs: [
      {
        arrival: {
          airportCode: "ADB",
          airportName: "Izmir",
          time: "2017-04-14T09:10:00"
        },
        carrier: "Pegasus Airlines",
        departure: {
          airportCode: "ADA",
          airportName: "Adana",
          time: "2017-04-14T07:40:00"
        },
        duration: 90,
        flightCode: "PC 3011"
      }
    ],
    lowestPrice: 66.99,
    origin: {
      airportCode: "ADA",
      time: "2017-04-14T07:40:00"
    },
    pricingOptions: [
      {
        agent: "Pegasus Airlines",
        price: 66.99
      }
    ],
    stops: 0,
    stopsType: StopsType.direct
  }
]

export const itineraryWithMultipleCarriers = {
  carriers: [
    {
      code: "PC",
      displayCode: "PC",
      id: 1161,
      imgUrl: "http://s1.apideeplink.com/images/airlines/H9.png",
      name: "Pegasus Airlines"
    },
    {
      code: "ABC",
      displayCode: "ABC",
      id: 123,
      imgUrl: "http://s1.apideeplink.com/images/airlines/H9.png",
      name: "ABC Airlines"
    }
  ],

  destination: {
    airportCode: "ADB",
    time: "2017-04-14T09:10:00"
  },
  duration: 90,
  id: "9249-1704140740--31973-0-9250-1704140910",
  legs: [
    {
      arrival: {
        airportCode: "ADB",
        airportName: "Izmir",
        time: "2017-04-14T09:10:00"
      },
      carrier: "Pegasus Airlines",
      departure: {
        airportCode: "ADA",
        airportName: "Adana",
        time: "2017-04-14T07:40:00"
      },
      duration: 90,
      flightCode: "PC 3011"
    }
  ],
  lowestPrice: 66.99,
  origin: {
    airportCode: "ADA",
    time: "2017-04-14T07:40:00"
  },
  pricingOptions: [
    {
      agent: "Pegasus Airlines",
      price: 66.99
    }
  ],
  stops: 0,
  stopsType: StopsType.direct
}
