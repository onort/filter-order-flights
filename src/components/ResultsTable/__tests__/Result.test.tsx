import React from "react"
import { mount, shallow } from "enzyme"

import { Result } from "../"
import testData, { itineraryWithMultipleCarriers } from "../../../data/testData"

describe("<Result />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<Result itinerary={testData[0]} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should toggle details when button is clicked", () => {
    const wrapper = mount(<Result itinerary={testData[0]} />)
    expect(wrapper.find("ResultDetails").exists()).toBe(false)
    wrapper.find("Button").simulate("click")
    expect(wrapper.find("ResultDetails").exists()).toBe(true)
    wrapper.find("Button").simulate("click")
    expect(wrapper.find("ResultDetails").exists()).toBe(false)
    wrapper.unmount()
  })

  it("should have carrier names instead of logos if there are more than one carriers", () => {
    const wrapper = shallow(
      <Result itinerary={itineraryWithMultipleCarriers} />
    )
    expect(wrapper.find(".logo").exists()).toBe(false)
    expect(wrapper.find(".carrierName")).toHaveLength(2)
  })

  it("should have multiple airlines text in priceCell if there are more than one carriers", () => {
    const wrapper = mount(<Result itinerary={itineraryWithMultipleCarriers} />)
    expect(wrapper.find(".price .dataCell .sub").text()).toBe(
      "Farklı havayolları"
    )
    wrapper.unmount()
  })
})
