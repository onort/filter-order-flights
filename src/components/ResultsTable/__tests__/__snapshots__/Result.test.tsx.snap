// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Result /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Result
    itinerary={
      Object {
        "carriers": Array [
          Object {
            "code": "PC",
            "displayCode": "PC",
            "id": 1161,
            "imgUrl": "http://s1.apideeplink.com/images/airlines/H9.png",
            "name": "Pegasus Airlines",
          },
        ],
        "destination": Object {
          "airportCode": "ADB",
          "time": "2017-04-14T09:10:00",
        },
        "duration": 90,
        "id": "9249-1704140740--31973-0-9250-1704140910",
        "legs": Array [
          Object {
            "arrival": Object {
              "airportCode": "ADB",
              "airportName": "Izmir",
              "time": "2017-04-14T09:10:00",
            },
            "carrier": "Pegasus Airlines",
            "departure": Object {
              "airportCode": "ADA",
              "airportName": "Adana",
              "time": "2017-04-14T07:40:00",
            },
            "duration": 90,
            "flightCode": "PC 3011",
          },
        ],
        "lowestPrice": 66.99,
        "origin": Object {
          "airportCode": "ADA",
          "time": "2017-04-14T07:40:00",
        },
        "pricingOptions": Array [
          Object {
            "agent": "Pegasus Airlines",
            "price": 66.99,
          },
        ],
        "stops": 0,
        "stopsType": "direct",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Row
          className="row"
        >
          <Cell
            className="carrierCell"
          >
            <img
              alt="Pegasus Airlines Logo"
              className="logo"
              src="http://s1.apideeplink.com/images/airlines/H9.png"
            />
          </Cell>
          <DataCell
            className="origin"
            detailCell={false}
            sub="ADA"
            val="07:40"
          >
            <img
              alt="Plane Icon"
              className="plane"
              src="plane.svg"
            />
          </DataCell>
          <DataCell
            detailCell={false}
            sub="ADB"
            val="09:10"
          />
          <DataCell
            className="stopsCell"
            detailCell={false}
            sub="1 saat 30 dakika"
            val="Direkt"
          />
          <Cell
            className="priceCell"
          >
            <div
              className="price"
            >
              <div
                className="dataCell"
              >
                <span
                  className="val"
                >
                  66.99
                   TL
                </span>
                <span
                  className="sub"
                >
                  Pegasus Airlines
                </span>
              </div>
              <div
                className="buttonContainer"
              >
                <Button
                  className="viewButton"
                  onClick={[Function]}
                  text="İncele"
                  type="button"
                />
              </div>
            </div>
          </Cell>
        </Row>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Cell
              className="carrierCell"
            >
              <img
                alt="Pegasus Airlines Logo"
                className="logo"
                src="http://s1.apideeplink.com/images/airlines/H9.png"
              />
            </Cell>,
            <DataCell
              className="origin"
              detailCell={false}
              sub="ADA"
              val="07:40"
            >
              <img
                alt="Plane Icon"
                className="plane"
                src="plane.svg"
              />
            </DataCell>,
            <DataCell
              detailCell={false}
              sub="ADB"
              val="09:10"
            />,
            <DataCell
              className="stopsCell"
              detailCell={false}
              sub="1 saat 30 dakika"
              val="Direkt"
            />,
            <Cell
              className="priceCell"
            >
              <div
                className="price"
              >
                <div
                  className="dataCell"
                >
                  <span
                    className="val"
                  >
                    66.99
                     TL
                  </span>
                  <span
                    className="sub"
                  >
                    Pegasus Airlines
                  </span>
                </div>
                <div
                  className="buttonContainer"
                >
                  <Button
                    className="viewButton"
                    onClick={[Function]}
                    text="İncele"
                    type="button"
                  />
                </div>
              </div>
            </Cell>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <img
                alt="Pegasus Airlines Logo"
                className="logo"
                src="http://s1.apideeplink.com/images/airlines/H9.png"
              />,
              "className": "carrierCell",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Pegasus Airlines Logo",
                "className": "logo",
                "src": "http://s1.apideeplink.com/images/airlines/H9.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <img
                alt="Plane Icon"
                className="plane"
                src="plane.svg"
              />,
              "className": "origin",
              "detailCell": false,
              "sub": "ADA",
              "val": "07:40",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Plane Icon",
                "className": "plane",
                "src": "plane.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "detailCell": false,
              "sub": "ADB",
              "val": "09:10",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "stopsCell",
              "detailCell": false,
              "sub": "1 saat 30 dakika",
              "val": "Direkt",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <div
                className="price"
              >
                <div
                  className="dataCell"
                >
                  <span
                    className="val"
                  >
                    66.99
                     TL
                  </span>
                  <span
                    className="sub"
                  >
                    Pegasus Airlines
                  </span>
                </div>
                <div
                  className="buttonContainer"
                >
                  <Button
                    className="viewButton"
                    onClick={[Function]}
                    text="İncele"
                    type="button"
                  />
                </div>
              </div>,
              "className": "priceCell",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="dataCell"
                  >
                    <span
                      className="val"
                    >
                      66.99
                       TL
                    </span>
                    <span
                      className="sub"
                    >
                      Pegasus Airlines
                    </span>
                  </div>,
                  <div
                    className="buttonContainer"
                  >
                    <Button
                      className="viewButton"
                      onClick={[Function]}
                      text="İncele"
                      type="button"
                    />
                  </div>,
                ],
                "className": "price",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span
                        className="val"
                      >
                        66.99
                         TL
                      </span>,
                      <span
                        className="sub"
                      >
                        Pegasus Airlines
                      </span>,
                    ],
                    "className": "dataCell",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          66.99,
                          " TL",
                        ],
                        "className": "val",
                      },
                      "ref": null,
                      "rendered": Array [
                        66.99,
                        " TL",
                      ],
                      "type": "span",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Pegasus Airlines",
                        "className": "sub",
                      },
                      "ref": null,
                      "rendered": "Pegasus Airlines",
                      "type": "span",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Button
                      className="viewButton"
                      onClick={[Function]}
                      text="İncele"
                      type="button"
                    />,
                    "className": "buttonContainer",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "className": "viewButton",
                      "onClick": [Function],
                      "text": "İncele",
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      false,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Row
            className="row"
          >
            <Cell
              className="carrierCell"
            >
              <img
                alt="Pegasus Airlines Logo"
                className="logo"
                src="http://s1.apideeplink.com/images/airlines/H9.png"
              />
            </Cell>
            <DataCell
              className="origin"
              detailCell={false}
              sub="ADA"
              val="07:40"
            >
              <img
                alt="Plane Icon"
                className="plane"
                src="plane.svg"
              />
            </DataCell>
            <DataCell
              detailCell={false}
              sub="ADB"
              val="09:10"
            />
            <DataCell
              className="stopsCell"
              detailCell={false}
              sub="1 saat 30 dakika"
              val="Direkt"
            />
            <Cell
              className="priceCell"
            >
              <div
                className="price"
              >
                <div
                  className="dataCell"
                >
                  <span
                    className="val"
                  >
                    66.99
                     TL
                  </span>
                  <span
                    className="sub"
                  >
                    Pegasus Airlines
                  </span>
                </div>
                <div
                  className="buttonContainer"
                >
                  <Button
                    className="viewButton"
                    onClick={[Function]}
                    text="İncele"
                    type="button"
                  />
                </div>
              </div>
            </Cell>
          </Row>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Cell
                className="carrierCell"
              >
                <img
                  alt="Pegasus Airlines Logo"
                  className="logo"
                  src="http://s1.apideeplink.com/images/airlines/H9.png"
                />
              </Cell>,
              <DataCell
                className="origin"
                detailCell={false}
                sub="ADA"
                val="07:40"
              >
                <img
                  alt="Plane Icon"
                  className="plane"
                  src="plane.svg"
                />
              </DataCell>,
              <DataCell
                detailCell={false}
                sub="ADB"
                val="09:10"
              />,
              <DataCell
                className="stopsCell"
                detailCell={false}
                sub="1 saat 30 dakika"
                val="Direkt"
              />,
              <Cell
                className="priceCell"
              >
                <div
                  className="price"
                >
                  <div
                    className="dataCell"
                  >
                    <span
                      className="val"
                    >
                      66.99
                       TL
                    </span>
                    <span
                      className="sub"
                    >
                      Pegasus Airlines
                    </span>
                  </div>
                  <div
                    className="buttonContainer"
                  >
                    <Button
                      className="viewButton"
                      onClick={[Function]}
                      text="İncele"
                      type="button"
                    />
                  </div>
                </div>
              </Cell>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <img
                  alt="Pegasus Airlines Logo"
                  className="logo"
                  src="http://s1.apideeplink.com/images/airlines/H9.png"
                />,
                "className": "carrierCell",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Pegasus Airlines Logo",
                  "className": "logo",
                  "src": "http://s1.apideeplink.com/images/airlines/H9.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <img
                  alt="Plane Icon"
                  className="plane"
                  src="plane.svg"
                />,
                "className": "origin",
                "detailCell": false,
                "sub": "ADA",
                "val": "07:40",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Plane Icon",
                  "className": "plane",
                  "src": "plane.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "detailCell": false,
                "sub": "ADB",
                "val": "09:10",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "stopsCell",
                "detailCell": false,
                "sub": "1 saat 30 dakika",
                "val": "Direkt",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <div
                  className="price"
                >
                  <div
                    className="dataCell"
                  >
                    <span
                      className="val"
                    >
                      66.99
                       TL
                    </span>
                    <span
                      className="sub"
                    >
                      Pegasus Airlines
                    </span>
                  </div>
                  <div
                    className="buttonContainer"
                  >
                    <Button
                      className="viewButton"
                      onClick={[Function]}
                      text="İncele"
                      type="button"
                    />
                  </div>
                </div>,
                "className": "priceCell",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="dataCell"
                    >
                      <span
                        className="val"
                      >
                        66.99
                         TL
                      </span>
                      <span
                        className="sub"
                      >
                        Pegasus Airlines
                      </span>
                    </div>,
                    <div
                      className="buttonContainer"
                    >
                      <Button
                        className="viewButton"
                        onClick={[Function]}
                        text="İncele"
                        type="button"
                      />
                    </div>,
                  ],
                  "className": "price",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <span
                          className="val"
                        >
                          66.99
                           TL
                        </span>,
                        <span
                          className="sub"
                        >
                          Pegasus Airlines
                        </span>,
                      ],
                      "className": "dataCell",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            66.99,
                            " TL",
                          ],
                          "className": "val",
                        },
                        "ref": null,
                        "rendered": Array [
                          66.99,
                          " TL",
                        ],
                        "type": "span",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Pegasus Airlines",
                          "className": "sub",
                        },
                        "ref": null,
                        "rendered": "Pegasus Airlines",
                        "type": "span",
                      },
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Button
                        className="viewButton"
                        onClick={[Function]}
                        text="İncele"
                        type="button"
                      />,
                      "className": "buttonContainer",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "className": "viewButton",
                        "onClick": [Function],
                        "text": "İncele",
                        "type": "button",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        false,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`<Result /> should match snapshot 2`] = `"<div class=\\"row row\\"><div class=\\"cell carrierCell\\"><img class=\\"logo\\" src=\\"http://s1.apideeplink.com/images/airlines/H9.png\\" alt=\\"Pegasus Airlines Logo\\"/></div><div class=\\"cell dataCell origin\\"><span class=\\"val\\">07:40</span><span class=\\"sub\\">ADA</span><img class=\\"plane\\" src=\\"plane.svg\\" alt=\\"Plane Icon\\"/></div><div class=\\"cell dataCell\\"><span class=\\"val\\">09:10</span><span class=\\"sub\\">ADB</span></div><div class=\\"cell dataCell stopsCell\\"><span class=\\"val\\">Direkt</span><span class=\\"sub\\">1 saat 30 dakika</span></div><div class=\\"cell priceCell\\"><div class=\\"price\\"><div class=\\"dataCell\\"><span class=\\"val\\">66.99 TL</span><span class=\\"sub\\">Pegasus Airlines</span></div><div class=\\"buttonContainer\\"><button class=\\"button viewButton\\" type=\\"button\\"><span class=\\"text\\">İncele</span></button></div></div></div></div>"`;
